### PR TITLE
Clean up command perms

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR

revival of #20236

- adds new Moderator perm for moderation actions
- adds new adminwho perm for using adminwho
- adds new namecolor perm for changing ooc name color
- moves a lot of previous `ADMIN` flagged commands to `MODERATOR` and `FUN`.

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: ShadowCommander, EmoGarbage
ADMIN:
- add: Added new permissions: moderator, adminwho, and namecolor
- tweak: Changed various command perms from admin to moderator & fun.

### CODE REVIEW

**Comment by mirrorcult** [comment]
does this have admin team approval

File: Content.Server/Chat/Commands/SetOOCCommand.cs [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 namespace Content.Server.Chat.Commands;
 
-[AdminCommand(AdminFlags.Server)]
+[AdminCommand(AdminFlags.NameColor)]
```
[the remark this answers is not in the document]
Comment (EmoGarbage404) [reply Content.Server/Chat/Commands/SetOOCCommand.cs:9]: Mirino lied to me I'm going to have to shatter him into a million atoms

**Comment by EmoGarbage404** [comment]
> does this have admin team approval

this is based off of [chief engineer's comment](https://github.com/space-wizards/space-station-14/pull/20236#issuecomment-1722178464) from shadowcommander's original pr.

**Comment by moonheart08** [comment]
> > does this have admin team approval
> 
> as far as I can tell, assuming game admins are given moderator, adminwho, and namecolor after it's merged, nothing changes in terms of admin capabilities. I don't really remember the specifics of why some maintainers wanted the `ADMIN` perm in-game, but I'm assuming this is related to that

+ADMIN gates a lot of basic functionality like aghost and *adminchat*, making coordination for debugging/etc impossible without it.

**Comment by PJB3005** [comment]
We should just code a DB migration to add new flags automatically.

**Review by ShadowCommander — APPROVED** [review]

**Review by ShadowCommander — APPROVED** [review]

**Pull request merged**